### DISCUSSION TITLE
crazyflie-firmware for Cortex R5 (Zynq Ultrascale+ platform)

### DISCUSSION THREAD

**Discussion by jostmi** [bitcraze/discussions#677] [Q&A] [answered]
I am about to port the crazyflie-firmware to a Cortex R5 processor (part of the Zynq Ultrascale+ platform).

Has anyone done that already?
Or ported the software to another similar processor?

**Comment by ataffanel** [comment, marked as answer]
The Crazyflie firmware has never been ported on this specific processor as far as I know.

Though, we have been porting it to a couple of processor in the past, for once Crazyflie 1 and Crazyflie 2 have been running the same firmware branch for a while and it was 2 quite different processor, this was before we used the KBuild build system however. All the  processor we have tested though where of the Cortex-M family, we have never worked with a Cortex-R. As long as you have gcc for your target processor, any should work fine though. The main work will be to get the driver to work.

The firmware has been designed to be quite portable but this design has not been tested a lot lately. In theory the src/HAL and src/drivers source folder should be ported to a new platform, while the src/module folder should be platform independent.

We are currently supporting multiple platforms via the KBuild build system but all of them have the same processor. My assumption is that the first things work on will be to configure KBuild to compile using, potentially, another compiler and other linker scripts, and to use the right FreeRTOS port.
